appserver/web/web-core/src/main/java/org/apache/catalina/connector/CoyoteAdapter.java

```diff
@@ -132,6 +133,8 @@ public class CoyoteAdapter extends HttpHandler {
         Boolean.valueOf(System.getProperty(
             "com.sun.enterprise.web.collapseAdjacentSlashes", "true"));
 
+    private static String INVALID_PATTERNS_RFC_9110 = "(\\\\n)|(\\\\0)|(\\\\r)|(\\\\x00)|(\\\\x0A)|(\\\\x0D)";
```

Comment:
There's no limitation to literal `\` characters in RFC9110.

The actual bytes that involve non-visual characters should be rejected (`Character.isISOControl`), as actuall `0x00` in the byte stream.